cmd/aro/operator.go

```diff
@@ -76,54 +78,93 @@ func operator(ctx context.Context, log *logrus.Entry) error {
+			Kubernetescli: kubernetescli,
+			Securitycli:   securitycli,
+			AROCli:        arocli,
+			RestConfig:    restConfig,
+			Log:           log.WithField("controller", "Genevalogging"),
+			Scheme:        mgr.GetScheme(),
+		}).SetupWithManager(mgr); err != nil {
+			log.Errorf("unable to create controller: Genevalogging %v", err)
+			return err
+		}
+		if err = (&controllers.PullsecretReconciler{
+			Kubernetescli: kubernetescli,
+			AROCli:        arocli,
+			Log:           log.WithField("controller", "PullSecret"),
+			Scheme:        mgr.GetScheme(),
+		}).SetupWithManager(mgr); err != nil {
+			log.Errorf("unable to create controller: PullSecret %v", err)
+			return err
+		}
+		if err = (&controllers.AlertWebhookReconciler{
+			Kubernetescli: kubernetescli,
+			Log:           log.WithField("controller", "AlertWebhook"),
+			Scheme:        mgr.GetScheme(),
+		}).SetupWithManager(mgr); err != nil {
+			log.Errorf("unable to create controller: AlertWebhook %v", err)
+			return err
+		}
+		if err = (&controllers.CPValidator{
+			Kubernetescli: kubernetescli,
+			AROCli:        arocli,
+			Log:           log.WithField("controller", "CPValidator"),
+			Scheme:        mgr.GetScheme(),
+		}).SetupWithManager(mgr); err != nil {
+			log.Errorf("unable to create controller: CPValidator %v", err)
+			return err
+		}
+		if err = (&controllers.AlertWebhookReconciler{
+			Kubernetescli: kubernetescli,
+			Log:           log.WithField("controller", "AlertWebhook"),
+			Scheme:        mgr.GetScheme(),
+		}).SetupWithManager(mgr); err != nil {
+			log.Errorf("unable to create controller: AlertWebhook %v", err)
+			return err
+		}
 	}
+
 	if err = (&controllers.InternetChecker{
 		Kubernetescli: kubernetescli,
 		AROCli:        arocli,
 		Log:           log.WithField("controller", "InternetChecker"),
 		Scheme:        mgr.GetScheme(),
+		Placement:     placement,
 	}).SetupWithManager(mgr); err != nil {
 		log.Errorf("unable to create controller: InternetChecker %v", err)
 		return err
 	}
-	if err = (&controllers.CPValidator{
-		Kubernetescli: kubernetescli,
-		AROCli:        arocli,
-		Log:           log.WithField("controller", "CPValidator"),
-		Scheme:        mgr.GetScheme(),
-	}).SetupWithManager(mgr); err != nil {
-		log.Errorf("unable to create controller: CPValidator %v", err)
-		return err
-	}
-	if err = (&controllers.AlertWebhookReconciler{
-		Kubernetescli: kubernetescli,
-		Log:           log.WithField("controller", "AlertWebhook"),
-		Scheme:        mgr.GetScheme(),
-	}).SetupWithManager(mgr); err != nil {
-		log.Errorf("unable to create controller: AlertWebhook %v", err)
-		return err
-	}
+
 	// +kubebuilder:scaffold:builder
 
 	log.Info("starting manager")
 	return mgr.Start(ctrl.SetupSignalHandler())
 }
+
+// Parses `nodeName` environment variable to retrieve pod's placement - master or worker
+func getPlacement(log *logrus.Entry) (string, error) {
+	nodeName, found := os.LookupEnv("NODE_NAME")
```

Comment:
maybe also cater for "go run ./cmd/aro operator" - if RP_MODE=development then assume master